src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = {...state};
+  let actionHistory = [];
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateClone = {...stateClone,...action.extraData}
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          delete stateClone[key];
+        });
+        break;
+      case 'clear':
+        stateClone = {};
+
+        break;
+      default:
```

Comment:
Consider adding a `break` statement in the `default` case of the switch statement. Although not strictly necessary here, it's a good practice to include it for consistency and to prevent potential fall-through issues if more cases are added in the future.